### DISCUSSION TITLE
Unable to run fresh install on fresh install of xampp 8.2.12

### DISCUSSION THREAD

**Discussion by sausje** [chevereto/chevereto#98] [Q&A]
I keep getting the error that something went wrong before I can even install. 

The apache logs show the following:

```
[6751ae7913711] RuntimeException in C:\xampp\htdocs\Chevereto\app\src\Legacy\functions.php:1228

# Message [Code #600]
Sessions not working (session_start)

# URI
/chevereto/install

# Method
GET

# Time
2024-12-05T13:45:29+00:00 [1733406329]

# Incident 6751ae7913711

# Backtrace
------------------------------------------------------------
0 C:\xampp\htdocs\Chevereto\app\src\Legacy\functions.php:1228
{main}()
------------------------------------------------------------
1 C:\xampp\htdocs\Chevereto\app\legacy\entrypoints\index.php:37
Chevereto\Legacy\loaderHandler(array(size=1), array(size=0), array(size=0), array(size=0), array(size=0), array(size=0), array(size=56), array(size=0))
------------------------------------------------------------
2 C:\xampp\htdocs\Chevereto\index.php:12
require_once(string(length=58))
------------------------------------------------------------

```

The rewrite module is on and afaik all permissions should be fine. Is there something I'm missing?

**Comment by rodber** [comment]
By default it uses `/tmp` for sessions. You can either fix permissions there for the user running PHP, or you can tweak the [Chevereto session variables](https://v4-docs.chevereto.com/application/configuration/environment.html#session-variables) to use a different path where you can setup the right permissions. You can also use the Redis handler.

**Reply by sausje** [reply]
Thanks for the quick reply.

I've tried setting permissions for the /tmp folder in the php dir but that didn't seem to work. (I presume that's the tmp folder being used?)
So I tried setting the session_save_handler to redis but that also results in the following error:

```
[6751b63a6db3d] ErrorException in C:\xampp\htdocs\Chevereto\app\src\Legacy\functions.php:1160

# Message [Code #2]
ini_set(): Session save handler &quot;redis&quot; cannot be found

# URI


# Method


# Time
2024-12-05T14:18:34+00:00 [1733408314]

# Incident 6751b63a6db3d

# Backtrace
------------------------------------------------------------
0 C:\xampp\htdocs\Chevereto\app\src\Legacy\functions.php:1160
{main}()
------------------------------------------------------------
1 C:\xampp\htdocs\Chevereto\app\src\Legacy\functions.php:1160
ini_set(string(length=20), string(length=5))
------------------------------------------------------------
2 C:\xampp\htdocs\Chevereto\app\legacy\entrypoints\index.php:37
Chevereto\Legacy\loaderHandler(array(size=1), array(size=0), array(size=0), array(size=0), array(size=0), array(size=0), array(size=57), array(size=0))
------------------------------------------------------------
3 C:\xampp\htdocs\Chevereto\index.php:12
require_once(string(length=58))
------------------------------------------------------------

```

Update: So that error is because redis isn't installed by default on xampp. I've added it now and confirmed it's there via phpinfo(), and now it's throwing the sessions not working error again.

**Comment by sausje** [comment]
Update: it was because it refers to /tmp while in xampp it's default php tmp dir is /xampp/tmp (as it's specified in the php values).

However now I've got an issue with the following:

```
[675d9269a7e7d] LogicException in D:\xampp\htdocs\app\src\Legacy\G\Handler.php:107

# Message 
Invalid PHP front controller setup. Review your web server configuration.

# URI
/

# Method
GET

# Time
2024-12-14T14:12:57+00:00 [1734185577]

# Incident 675d9269a7e7d

# Backtrace
------------------------------------------------------------
0 D:\xampp\htdocs\app\src\Legacy\G\Handler.php:107
{main}()
------------------------------------------------------------
1 D:\xampp\htdocs\app\legacy\load\web.php:662
Chevereto\Legacy\G\Handler->__construct(bool(true), Closure(#18), Closure(#15))
------------------------------------------------------------
2 D:\xampp\htdocs\app\legacy\entrypoints\index.php:45
require_once(string(length=39))
------------------------------------------------------------
3 D:\xampp\htdocs\index.php:12
require_once(string(length=48))
------------------------------------------------------------
```

Which I gathered has to do with the mod_rewrite settings?
I've enabled the rewrite module and have set the correct settings for directory in httpd.conf: https://v4-docs.chevereto.com/application/stack/web-server.html#apache-http-server

```
<Directory />
	Options -Indexes +FollowSymLinks +MultiViews
	AllowOverride All
	Require all granted
</Directory>
```

(I've also tried the same but with directory /xampp/htdocs)

**Comment by sausje** [comment]
Seems like even other users from XAMPP trying to set this up in a test environment are failing (even the mod on the Apache forums). So this is an issue with Chevereto V4 not being able to run properly on XAMPP (Apache). 

Quote from this person:

> I have tried the installation myself on my test environment and failed.
> It IS a third party application problem, because by default XAMPP supports sessions and front controller patterns by mod_rewrite.
> By default sessions are active and mod_rewrite too for the htdocs folder. No additional configuration needed.
> By default XAMPP has a tmp folder it uses. Just chevereto has its own configuration for a tmp folder, without any need.

Could you please look into this.

**Comment by rodber** [comment]
It seems the issue could be related to the underlying Windows filesystem, as [Chevereto is built and tested on Unix-like systems](https://v4-docs.chevereto.com/application/stack/operating-system.html). We don't have a Windows team, we rely on users like you exposing these issues.

I understand you're using XAMPP for local testing/development, but keep in mind that in a production environment, Chevereto would typically run on Unix-like systems.

One potential issue I can think of is related to this code snippet: https://github.com/chevereto/chevereto/blob/ed56a10e50b9bd92f967be5912b6015b3f9df1a7/app/src/Legacy/G/Handler.php#L102

Chevereto checks for a valid PHP setup by rejecting direct access to PHP files. It performs a check using `file_exists`, and either the Windows API might incorrectly return `true` for `/`, or it might cause an error that's being suppressed.

I suggest you try modifying the code as follows to reveal the actual error:

Change this at `app/src/Legacy/G/Handler.php`

```php
            try {
                $requestFileExists = file_exists($pathRequest);
            } catch (Throwable $e) {
                $requestFileExists = true;
            }
```

To this:

```php
            //try {
                $requestFileExists = file_exists($pathRequest);
            //} catch (Throwable $e) {
            //    $requestFileExists = true;
            //}
```

This change will surface the error (if any).

**Reply by sausje** [reply]
Sadly that didn't show anything. It's still the same error with "Invalid PHP front controller setup." in the Apache error.log.

Sadly I'm not good with Unix and that's why I run my little home server with Windows. This one is just on an N100 with 16GB of memory, so running a docker is also impossible and would be way too slow. 
So whatever is needed, I'll gladly help to find out how we can make this work on Windows systems. It's just for me personally hosted, since we still use teamspeak rather than Discord, so using that for easy image sharing to my friends. (Rather than relying on 3rd party, I rather selfhost things like this.)

Anyway, another odd thing which might help narrow down, is that the above error shows if it's in the root of XAMPP (htdocs). Once I move it to, for example, /chevereto, the page kinda loads but fails to fully load. Console showing this:

![Screenshot (2)](https://github.com/user-attachments/assets/eaf496ae-7369-44d3-b50c-1ff7b2552e51)
